packages/run-protocol/src/psm/psm.js

```diff
@@ -0,0 +1,170 @@
+// @ts-check
+import '@agoric/zoe/exported.js';
+import '@agoric/zoe/src/contracts/exported.js';
+import '@agoric/governance/src/exported';
+import { assertProposalShape } from '@agoric/zoe/src/contractSupport/index.js';
+import {
+  ceilMultiplyBy,
+  makeRatio,
+} from '@agoric/zoe/src/contractSupport/ratio.js';
+import { Far } from '@endo/marshal';
+// import { CONTRACT_ELECTORATE } from '@agoric/governance';
+
+import { AmountMath } from '@agoric/ertp';
+import { makeMakeCollectFeesInvitation } from '../vaultFactory/collectRewardFees.js';
+
+const { details: X } = assert;
+
+const BASIS_POINTS = 10000n;
+
+/**
+ * Stage a transfer of a singe asset from one seat to another, with an optional
+ * remapping of the Keywords. Check that the remapping is for the same amount.
+ *
+ * @param {ZCFSeat} from
+ * @param {ZCFSeat} to
+ * @param {AmountKeywordRecord} txFrom
+ * @param {AmountKeywordRecord} txTo
+ */
+function stageTransfer(from, to, txFrom, txTo = txFrom) {
+  assert(AmountMath.isEqual(Object.values(txFrom)[0], Object.values(txTo)[0]));
+  from.decrementBy(txFrom);
+  to.incrementBy(txTo);
+}
+
+/**
+ * @param {ZCF<{
+ *    anchorBrand: Brand,
+ *    main: {
+ *      WantStableFeeBP: bigint,
+ *      GiveStableFeeBP: bigint,
+ *      MintLimit: Amount } }>
```

Comment:
I think we also need a multiplication factor here. Even if we're trading at "par", the different currencies may have a different number of decimal places.

Reply:
I added it to terms rather than governance